### DISCUSSION TITLE
I, Voyager needs to facilitate building planets, moons, spaceships, solar systems, etc., via Godot Editor, not just via data tables

### DISCUSSION THREAD

**Discussion by charliewhitfield** [ivoyager/forum#9] [General]
I feel like this is a major problem for many or most potential developers, and I'm adding it to the [Roadmap](https://github.com/orgs/ivoyager/discussions/5). However, I'm probably the least qualified developer to solve it.

Some background/context:

I, Voyager builds *everything* by code from *.tsv data tables (stars, planets, moons, spacecrafts) or binaries (asteroids). Almost all resources (models, textures) and binaries are specified in the data tables, usually by a file name prefix rather than the whole path (the resource files are found and loaded at runtime). This is powerful and exactly what I needed for the Planetarium and [my own game](https://t2civ.com/) project.

However, I guess that most game developers would expect to build their system in the Godot Editor. *Add Body. Adjust mass, radius, luminosity, color, etc., to make it a star. Add another Body to the first. Add Texture(s), Orbit, etc., and set Body and Orbit parameters to make it a planet in the desired orbit...* And so on, all from the editor just like building a dungeon or whatever. This may seem obvious to you but wasn't to me until recently. Currently, there aren't even any `export` variables anywhere in our code.

Sadly, this is not something that I will have the time to do myself. Also, I've never even built a 3D scene in the Godot Editor or any other editor (I'm basically a data programmer new to game development). However, I'm very willing to facilitate this effort in any way I can if someone else takes the lead. Or if you have structural/coding suggestions that might take us in the right direction, please do post here.

Any changes of course will have to preserve the existing build-by-data-table system.

**Comment by 0xIonRod** [comment]
That's exactly what I'm working on!

I'm thinking of relying on [GMAT](https://software.nasa.gov/software/GSC-17177-1) as a basis for UI. Kind of building a better UI for GMAT.

I'll share my thoughts once I'll have more specific ideas
